cozmo_soar.py

```diff
@@ -648,6 +654,153 @@ def on_input_phase(self, input_link: sml.Identifier):
+                t4 = +1.0 - 2.0 * (y * y + z * z)
+                Z = math.atan2(t3, t4)
+                return (X, Y, Z)
+
+            eul = quaternion_to_euler(quat)
+
+            if type(obj).__name__ == '_proxy_LightCube':
+                LIGHT_CUBE_LENGTH = 43
+                depth = LIGHT_CUBE_LENGTH
+                width = LIGHT_CUBE_LENGTH
+                height = LIGHT_CUBE_LENGTH
+            
+            else:
+                depth = obj.x_size_mm
+                width = obj.y_size_mm
+                height = obj.z_size_mm
+
+            # generate [GEOMETRY] for svs str by generating vertices from position and vectors of dimensions      
+            def combinations(possibilities):
+                """
+                Takes in array of the form [(tuple), (tuple), (tuple)] and returns all the possible combinations.  You can think of each tuple as possibilities of a value at a certain place (index in top level array).  This returns all possible combinations of those values in each place.
+                """
+                # will be modified during recursion
+                combs = []
+                def comb_helper(arr, i, baton):
+                    """
+                    Modifies combs variable above
+                    """
+
+                    for elt in arr[i]:
+                        baton.append(elt)
+                        if i < len(arr) - 1:
+                            comb_helper(arr, i + 1, baton)
+                        else:
+                            combs.append(copy.deepcopy(baton))
+                        baton.pop()
+                comb_helper(possibilities, 0, [])
+                return combs
+
+            # vectors from center of object to vertices, unrotated
+            vectors = np.transpose(np.array(combinations([(depth/2.0, -depth/2.0), (width/2.0, -width/2.0), (height/2.0, -height/2.0)])))
+            rotation_matrix = transforms3d.euler.euler2mat(eul[0], eul[1], eul[2], axes='sxyz')
+            rotated_vectors = np.transpose(np.matmul(rotation_matrix, vectors))
+            
+            # rotated vertices = position array + rotated vectors
+            vertices = np.transpose(np.array([[ai + bi for ai, bi in zip(pos_arr, vector)] for vector in rotated_vectors]))
+            
+            # generate [TRANSFORM] portion of svs string
+            scalar = 1000.0   # turns navmap oordinates in mm to SVS coordinates in m
+            gen_transform = lambda pos, rot: f"p {pos[0]/scalar} {pos[1]/scalar} {pos[2]/scalar} r {rot[0]} {rot[1]} {rot[2]}"
+            geo_str = "v"
+            for vertex in vertices:
+                geo_str += f' {vertex[0]/scalar} {vertex[1]/scalar} {vertex[2]/scalar}'
+            svs_str = f"{geo_str} {gen_transform(pos_arr, eul)}"
+            
+            if str_type == 'add':
+                return f'add {oid} world {svs_str}'
+            else:
+                return f'change {oid} {svs_str}'
+
+        def send_input(input):
```

Comment:
I recommend adding clarity to the function name by specifying it's SVS input. I suggest `send_svs_input`. Don't use `input` as a variable name, it's reserved to the built-in function `input`. I suggest something like `svs_in` which adds a bit of clarity as well.

```suggestion
        def send_svs_input(svs_in):
```